app/src/pages/Competition.tsx

```diff
@@ -0,0 +1,153 @@
+
+  useEffect(() => {
+    const getData = async () => {
+      const [compData, wcifData] = await Promise.all([
+        fetch(LINKS.WCA.API.COMPETITION_INFO + compid).then((response) => {
+          if (!response.ok) {
+            setHasError(true);
+          }
+          return response.json();
+        }),
+        fetch(LINKS.WCA.API.COMPETITION_INFO + compid + "/wcif/public").then(
+          (response) => {
+            if (!response.ok) {
+              setHasError(true);
+            }
+            return response.json();
+          },
+        ),
+      ]);
+
+      setCompetitionData({ ...compData, ...wcifData });
+      setIsLoading(false);
+    };
+    getData();
+  }, [compid]);
+
+  if (hasError) {
+    return (
+      <Container maxWidth="xl" style={{ textAlign: "center" }}>
+        <Box marginTop="4rem">
+          <Typography
+            component="h1"
+            variant="h5"
+            fontWeight="bold"
+            gutterBottom
+          >
+            {t("competition.error")}
+          </Typography>
+        </Box>
+      </Container>
+    );
+  }
+
+  if (competitionData.series) {
+    navigate(`/${locale}/competitions/series/${competitionData.series.id}`);
+  }
+
+  if (isLoading) {
+    return (
+      <Box sx={{ width: "100%" }}>
+        <LinearProgress />
+      </Box>
+    );
+  }
+
+  const competitorsApproved = (competition: any) => {
+    return competition.persons.filter((competitor: any) => {
+      return (
+        competitor.registration && competitor.registration.status === "accepted"
+      );
+    }).length;
+  };
+
+  const currentDate = new Date();
+  const registrationOpen = new Date(competitionData.registration_open);
+  const registrationClose = new Date(competitionData.registration_close);
+
+  return (
+    <Container maxWidth="xl" style={{ textAlign: "center" }}>
+      <Box marginTop="4rem">
+        <Typography component="h1" variant="h3" fontWeight="bold" gutterBottom>
+          {competitionData.name}
+        </Typography>
+        <Typography gutterBottom style={{ textAlign: "center" }}>
+          <Trans>
+            {currentDate < registrationOpen
+              ? t("competition.registration.before", { date: registrationOpen })
+              : t("competition.registration.after", { date: registrationOpen })}
+            {currentDate < registrationClose
+              ? t("competition.registration.closes", {
+                  date: registrationClose,
+                })
+              : t("competition.registration.closed", {
+                  date: registrationClose,
+                })}
+          </Trans>
+        </Typography>
+      </Box>
+      <Box display="flex" justifyContent="center" flexWrap="wrap">
+        <Box margin="1rem" padding="1rem">
+          <Typography gutterBottom>
+            <Trans>
+              {t("competition.date", {
+                date: new Date(
+                  competitionData.start_date + "T12:00:00.000Z",
```

Comment:
`start_date` only provides the day the competition starts, which `Date` then takes as midnight

Alternatively, we could parse the schedule within the wcif which contains more precise start and end times.

Reply:
So is the ` + "T12:00:00.000Z"` necessary? What's wrong with midnight?

Reply:
I forgot to mention that it interprets the date as midnight GMT, this means that the date shown for those in any Western time zone will be shown as the date before the competition without this correction. Setting the date to noon GMT should ensure everyone sees the same day regardless of the time zone.

Alternatively, as a time zone is provided in the wcif, I could change it so users will see the correct date in their time zone.

Reply:
Oh so `new Date` only interprets in the local timezone if it's a full date & time, but `start_date` is only a date, which means we need to add a time.

Why not add `"T00:00:00.000Z"`? Sticking with midnight seems simplest and least confusing (to me at least).

Reply:
The Date object does not store any time zone information, (the only value it actually stores is the number of milliseconds since 1970-01-01 00:00:00 UTC). Rather, all dates are parsed passed in with their time as an offset from UTC+00:00.

Thus, specifying `T00:00:00.000Z"` means that the date will be interpreted as midnight in GMT/UTC+00:00, which will result in the date being displayed being 1 day behind.

Adding 12 hours to specify noon works since at noon GMT, it will be the same day for everyone worldwide.

Another option would be to find the user's time zone offset, and then add it to the date when it's being parsed. This means that the date will be stored as midnight in the user's time zone.

Another way would be to use the time zone of the competition, however, due to how the time zone data data is encoded, there isn't any easy way that's guaranteed to work in all browsers.

There could be an entirely different solution that I'm missing however.

Reply:
Oops I meant `"T00:00:00.000"`, without the "Z". If you give `new Date` a date and time string, without a timezone, it will be interpreted in the user's zone.